### PR TITLE
Move Production Deployment to Argo

### PR DESCRIPTION
**Pull request by DeusAvalon**
Closes N/A

Adds the Release Pipeline for the Production Deployment to Argo and Cleansup the not needed `.drone.yml`.
Prod Release happen when creating a new `Release_tag` (same procedure as for our other Applications).

Argo App link: https://argocd.k8s.sidestream.tech/applications/argocd/auctions-ui-production?view=tree&resource=

Production URL: https://unified-auctions.makerdao.com/

Deployment Manifests: https://github.com/sidestream-tech/k8s-projects/tree/main/auctions-ui/production

Downtime of the Prod Environment during migration was **less then 1 min**.

Checklist:
- [x] issue number linked above after pound (`#`)
    - replace "Closes " with "Contributes to" or other if this PR does not close the issue
- [x] issue checkboxes are all addressed
- [x] manually checked my feature / not applicable
- [x] wrote tests / not applicable
- [x] attached screenshots / not applicable

### CODE REVIEW

File: .github/workflows/build-and-deploy-production.yml [hunk 1]
```diff
@@ -0,0 +1,146 @@
+name: Deploy Auctions UI production
+
+on:
+  push:
+    tags:
+      - '*'
```
Comment (valiafetisov) [inline .github/workflows/build-and-deploy-production.yml:6]: We previously used production release trigger, as it is done for electron app:
https://github.com/sidestream-tech/unified-auctions-ui/blob/3596555b94f73a3aea3d4481e2b82d4606d879d5/.github/workflows/build-electron-app.yaml#L3-L5

Why do you prefer to trigger it on tag creation instead? What's the main difference?

**Comment by valiafetisov** [comment]
Do you also want to remove [kubernetes](https://github.com/sidestream-tech/unified-auctions-ui/tree/main/kubernetes) folder from this repo or is still being used?

File: .github/workflows/build-and-deploy-production.yml (hunk 1 above)
Comment (DeusAvalon) [reply .github/workflows/build-and-deploy-production.yml:6]: No main difference as the outcome is the same for both :) I moved more to the `push on tag` approach as it allowed me to run test deployments with different tags. As i dont think i will do this on this project a lot its totally fine for me to use ur provided approach :D

**Comment by DeusAvalon** [comment]
Removed the kubernetes folder and updated the prod release trigger :)

**Review by valiafetisov — APPROVED** [review]

**Pull request merged**